spring-restdocs-core/src/main/java/org/springframework/restdocs/cli/CurlRequestSnippet.java

```diff
@@ -45,6 +44,8 @@
  */
 public class CurlRequestSnippet extends TemplatedSnippet {
 
+	private CommandFormatter commandFormatter = ConcatenatingCommandFormatter.newlineSeparator();
+
 	/**
 	 * Creates a new {@code CurlRequestSnippet} with no additional attributes.
 	 */
```

Comment:
OK, I understand now. I'm also in favour of putting the default formatter in one place, I just wasn't sure where should it be. After looking at the code again I think that `CliDocumentation` should be a good enough place to put it there.

But, to have it done in a clean way I would propose the following changes:

- `CurlRequestSnippet`: create a copy of all constructors with an additional `commandFormatter` parameter and deprecate the old ones
- `CliDocumentation`: create additional `curlRequest` method that takes `attributes` and `commandFormatter` parameters. Additionally, change all existing `CurlRequestSnippet` constructors so that they will receive the default formatter.
- `CurlRequestSnippetTests`: add the additional formatter parameter to each `CurlRequestSnippet` constructor or call the appropriate `CliDocumentation` method. The former option feels better since these are the `CurlRequestSnippet` tests.

I do understand that this is more or less the same that you have proposed in the first place but now it's written in one place and it's easier for me to see the whole picture.